### PR TITLE
Dynamic Controller config handling

### PR DESCRIPTION
**Pull request by Apollon77**
make Controller configuration to be applied dynamically

### CODE REVIEW

File: src/matter/ControllerNode.ts [hunk 1]
```diff
@@ -135,6 +117,30 @@ class Controller implements GeneralNode {
         });
     }
 
+    async applyConfiguration(config: MatterControllerConfig, isInitialization = false): Promise<MessageResponse> {
```
Comment (foxriver76) [inline src/matter/ControllerNode.ts:120]: we could add some JSDOC to newly created methods, what do you think?

**Review by foxriver76 — APPROVED** [review]
LGTM :+1:

File: src/matter/ControllerNode.ts (hunk 1 above)
Comment (Apollon77) [reply src/matter/ControllerNode.ts:120]: I put jsdoc generally on my list ,so separate PR, but agreed

File: src/main.ts [hunk 2]
```diff
@@ -696,59 +717,86 @@ export class MatterAdapter extends utils.Adapter {
...
+                    startkey: `${this.namespace}.bridges.`,
+                    endkey: `${this.namespace}.bridges.\u9999`,
+                },
+            );
+            bridgesObjects.rows.forEach(row => objects.push(row.value));
+        }
+
+        for (const object of objects) {
+            // No valid object or a sub-channel
+            if (!object || !object.native || object._id.split('.').length !== 4) return;
```
Comment (foxriver76) [inline src/main.ts:752]: return or continue?

File: src/main.ts [hunk 3]
```diff
@@ -696,59 +717,86 @@ export class MatterAdapter extends utils.Adapter {
...
-        for (const [bridgeId, bridge] of this.bridges.entries()) {
-            if (!bridges.find(obj => obj._id === bridgeId)) {
-                await bridge.stop();
-                this.bridges.delete(bridgeId);
+        // When we just handle one object we do not need to sync with running  devices and bridges
+        if (obj === undefined) {
+            // Objects existing, not deleted, so disable not enabled bridges or devices
+            for (const bridgeId of this.bridges.keys()) {
+                if (!bridges.find(obj => obj._id === bridgeId)) {
+                    this.log.info(`Bridge ${bridgeId} is not enabled anymore, so stop it.`);
```
Comment (foxriver76) [inline src/main.ts:784]: maybe the id in `"` again^^

File: src/main.ts [hunk 4]
```diff
@@ -696,59 +717,86 @@ export class MatterAdapter extends utils.Adapter {
...
+                    await this.stopBridgeOrDevice('bridge', bridgeId);
+                }
+            }
+            for (const deviceId of this.devices.keys()) {
+                if (!devices.find(obj => obj._id === deviceId)) {
+                    this.log.info(`Device ${deviceId} is not enabled anymore, so stop it.`);
```
Comment (foxriver76) [inline src/main.ts:790]: maybe the id in `"` again^^

File: src/main.ts [hunk 5]
```diff
@@ -763,23 +811,26 @@ export class MatterAdapter extends utils.Adapter {
 
                     this.bridges.set(bridge._id, matterBridge);
                 }
+            } else {
+                const config = await this.prepareMatterBridgeConfiguration(bridge.native as BridgeDescription);
+                if (config) {
+                    this.log.info(`Apply configuration update for bridge ${bridge._id}.`);
```
Comment (foxriver76) [inline src/main.ts:817]: maybe the id in `"` again^^

File: src/main.ts [hunk 6]
```diff
@@ -763,23 +811,26 @@ export class MatterAdapter extends utils.Adapter {
...
+                    await existingBridge.applyConfiguration(config);
+                } else {
+                    this.log.info(`Configuration for bridge ${bridge._id} is no longer valid or bridge disabled. Stopping it now.`);
```
Comment (foxriver76) [inline src/main.ts:820]: maybe the id in `"` again^^

**Review by foxriver76 — APPROVED** [review]
lgtm

File: src/main.ts (hunk 2 above)
Comment (Apollon77) [reply src/main.ts:752]: good catch

**Pull request merged**